plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubCreateRepositoryWebhook.ts

```diff
@@ -0,0 +1,176 @@
+}) {
+  const { integrations } = options;
+
+  const credentialsProviders = new Map(
+    integrations.github.list().map(integration => {
+      const provider = GithubCredentialsProvider.create(integration.config);
+      return [integration.config.host, provider];
+    }),
+  );
+
+  return createTemplateAction<{
+    repoUrl: string;
+    webhookUrl: string;
+    webhookSecretEnv?: string;
+    events?: string[];
+    active?: boolean;
+    contentType?: ContentType;
+    insecureSsl?: boolean;
+  }>({
+    id: 'github:repository:webhook:create',
+    description: 'Creates webhook for a repository on GitHub.',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl', 'webhookUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            description: `Accepts the format 'github.com?repo=reponame&owner=owner' where 'reponame' is the new repository name and 'owner' is an organization or username`,
+            type: 'string',
+          },
+          webhookUrl: {
+            title: 'Webhook URL',
+            description: 'The URL to which the payloads will be delivered',
+            type: 'string',
+          },
+          webhookSecretEnv: {
+            title: 'Environment variable containing Webhook Secret',
+            description: 'Name of the environment variable containing secret',
+            type: 'string',
+          },
+          events: {
+            title: 'Triggering Events',
+            description:
+              'Determines what events the hook is triggered for. Default: push',
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+          active: {
+            title: 'Active',
+            type: 'boolean',
+            description: `Determines if notifications are sent when the webhook is triggered. Default: true`,
+          },
+          contentType: {
+            title: 'Content Type',
+            type: 'string',
+            enum: ['form', 'json'],
+            description: `The media type used to serialize the payloads. The default is 'form'`,
+          },
+          insecureSsl: {
+            title: 'Insecure SSL',
+            type: 'boolean',
+            description: `Determines whether the SSL certificate of the host for url will be verified when delivering payloads. Default 'false'`,
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        repoUrl,
+        webhookUrl,
+        webhookSecretEnv,
+        events = ['push'],
+        active = true,
+        contentType = 'form',
+        insecureSsl = false,
+      } = ctx.input;
+
+      const { owner, repo, host } = parseRepoUrl(repoUrl, integrations);
+
+      if (!owner) {
+        throw new InputError(`No owner provided for repo ${repoUrl}`);
+      }
+
+      let secret;
+      if (webhookSecretEnv) {
+        secret = process.env[webhookSecretEnv];
```

Comment:
This is problematic as you could read any environment variable started with the process.
I'd vote for optionally passing the webhook secret as part of the action as with any other argument instead trying to read it from a user requested environment variable.

Reply:
I agree a secret could be a parameter from Scaffolder GUI, e.g.: `{{ parameters.webhookSecret }}`. 
However, I wanted to support a case where a user of a GUI does not know a secret and it has to be provided from a configuration. Then it is not safe to put it there explicitly. I made it environment variable name to support such a case. 
Maybe it would be the best to have a field: `webhookSecret` that would support all cases automatically. 

- `webhookSecret: {{ parameters.webhookSecret }}`
- `webhookSecret: plain_abga2si3rj5fjso8ajfncj`
- `webhookSecret: ${SECRET_ENV}`

Let me know what are your thoughts?
Furthermore, is there a standard way of env variables substitution? I did not notice that in Scaffolder template processing as it is in `app-config.yaml`?

Reply:
@freben I think, I missed your point in the implementation and I did not add default webhookSecret in `createGithubCreateRepositoryWebhookAction`. I was testing it e2e with a temporary copy of a webhook into local Backstage and I was able to make it work just through:
`const webhookAction = createGithubWebhookAction({integrations, webhookSecretDefault: process.env['BRAINLY_INTEGRATION_TOKEN']});`
Where I added one more function parameter to `createGithubWebhookAction` and then used it as a default in `handler()` ctx.input: `webhookSecret = webhookSecretDefault,`.

Could you confirm it is what you meant? 

I guess then I will need to provide this update in this action as well.

Reply:
Hm I think my initial idea was 

```ts
const webhookAction = createGithubWebhookAction({integrations, webhookSecret: process.env['BRAINLY_INTEGRATION_TOKEN']});
```

ie, no `default`, just plain THE webhook secret. Does that sound right? And then the action stores that as an actual variable value in code, and uses that when creating webhooks. No ctx involved at all

Reply:
Avoiding context does not allow it to use secrets provided from a GUI by a user. Unless I am missing something. Could you confirm?

Reply:
Yeah. I didn't note that that was a desired feature - would people commonly want to do that? But sure, then I agree.

Reply:
I am not certain it is going to be a very common case, however, I think it is wort to keep it as an option since the cost of implementation is negligible.

Reply:
I created a PR: https://github.com/backstage/backstage/pull/7028 to resolve this.